dotCMS/src/main/java/com/dotcms/health/checks/ShutdownHealthCheck.java

```diff
@@ -0,0 +1,140 @@
+package com.dotcms.health.checks;
+
+import com.dotcms.health.config.HealthCheckConfig.HealthCheckMode;
+import com.dotcms.health.model.HealthCheckResult;
+import com.dotcms.health.model.HealthStatus;
+import com.dotcms.health.util.HealthCheckBase;
+import com.dotcms.shutdown.ShutdownCoordinator;
+import com.dotmarketing.util.Config;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Health check that monitors system shutdown status for Kubernetes integration.
+ * 
+ * This check ensures proper traffic routing during shutdown:
+ * - READINESS: Fails immediately when shutdown begins to stop new traffic
+ * - LIVENESS: Continues to pass while system can still process existing requests
+ * 
+ * This allows for graceful shutdown with request draining:
+ * 1. Shutdown begins → /readyz fails → load balancer stops routing new traffic
+ * 2. Existing requests continue to be processed → /livez still passes
+ * 3. Once request draining completes → /livez can optionally fail for faster pod termination
+ * 
+ * Configuration Properties:
+ * - health.check.shutdown.mode = Safety mode (PRODUCTION, MONITOR_MODE, DISABLED)
+ * - health.check.shutdown.fail-liveness-on-completion = Whether to fail liveness after shutdown completes (default: false)
+ */
+public class ShutdownHealthCheck extends HealthCheckBase {
```

Comment:
we already have probes api for this, lets see if not doing the wheel twice

Reply:
This builds upon the new probe work here that is merged in now.   https://github.com/dotCMS/core/pull/32404.  We are in the process of deprecating the existing probes.   They are still in action while we test the impact of the changes.   We will be removing the current probes when we have fully switched over.